converters/google/fake_resource_data.go

```diff
@@ -12,85 +12,52 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-package tfplan
+// In order to interact with terraform-google-conversion, we need to be able to create
+// "terraform resource data" that supports a very limited subset of the API actually
+// used during the conversion process.
+package google
 
 import (
+	"fmt"
 	"strconv"
 	"strings"
 
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
-	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
 )
 
-// newFieldGetter accepts a resource schema map (field name --> Schema),
-// a state instance, and a diff and returns a type with field getter methods.
-func newFieldGetter(sch map[string]*schema.Schema, state *terraform.InstanceState, diff *terraform.InstanceDiff) *fieldGetter {
-	if state == nil && diff == nil {
-		panic("both state and diff should not be nil")
-	}
-
-	var baseState map[string]string
-	if state != nil {
-		baseState = state.Attributes
-	} else {
-		baseState = make(map[string]string)
-	}
-
-	stateReader := &schema.MapFieldReader{
-		Map:    schema.BasicMapReader(baseState),
-		Schema: sch,
-	}
-
-	var rdr schema.FieldReader
-	if diff == nil {
-		rdr = stateReader
-	} else {
-		rdr = &schema.DiffFieldReader{
-			Diff:   diff,
-			Source: stateReader,
-			Schema: sch,
-		}
-	}
-
-	return &fieldGetter{
-		rdr:    rdr,
-		schema: sch,
-		state:  state,
-	}
+// Compare to https://github.com/hashicorp/terraform-plugin-sdk/blob/97b4465/helper/schema/resource_data.go#L15
+type FakeResourceData struct {
+	reader schema.FieldReader
+	kind   string
+	schema map[string]*schema.Schema
 }
 
-// fieldGetter exposes the ability to get terraform resource instance
-// field values.
-type fieldGetter struct {
-	rdr    schema.FieldReader
-	schema map[string]*schema.Schema
-	state  *terraform.InstanceState
+// Kind returns the type of resource (i.e. "google_storage_bucket").
+func (d *FakeResourceData) Kind() string {
+	return d.kind
 }
 
 // Id returns the ID of the resource from state.
-func (g *fieldGetter) Id() string {
-	if g.state != nil {
```

Comment:
state was never actually set, so it was always nil. I removed it and this conditional for now; we can always implement this correctly later if needed.